File: app/src/main/java/com/rolandvitezhu/todocloud/ui/activity/main/dialogfragment/DatePickerDialogFragment.java
package com.rolandvitezhu.todocloud.ui.activity.main.dialogfragment;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.DatePicker;

import com.rolandvitezhu.todocloud.R;
import com.rolandvitezhu.todocloud.app.Constant;
import com.rolandvitezhu.todocloud.ui.activity.main.fragment.CreateTodoFragment;
import com.rolandvitezhu.todocloud.ui.activity.main.fragment.ModifyTodoFragment;

import org.threeten.bp.LocalDate;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatDialogFragment;

public class DatePickerDialogFragment extends AppCompatDialogFragment implements
    DatePickerDialog.OnDateSetListener {
	
	private int year;
	private int month;
	private int day;
  private LocalDate date;

  @NonNull
  @Override
  public Dialog onCreateDialog(Bundle savedInstanceState) {
    boolean thereIsNoDate = getArguments() == null || getArguments().get(Constant.DUE_DATE) == null;

    if (thereIsNoDate)
      date = LocalDate.now();
    else
      date = (LocalDate) getArguments().get(Constant.DUE_DATE);

    year = date.getYear();
    month = date.getMonthValue();
    day = date.getDayOfMonth();

	  return new DatePickerDialog(
        getActivity(), R.style.MyPickerDialogTheme, this, year, month - 1, day
    );
  }

  @Override
  public void onDateSet(DatePicker view, int year, int month, int day) {
    date = LocalDate.of(year, month + 1, day);

    if (getTargetFragment() instanceof CreateTodoFragment)
      ((CreateTodoFragment)getTargetFragment()).onSelectDate(date);
    else if (getTargetFragment() instanceof ModifyTodoFragment)
      ((ModifyTodoFragment)getTargetFragment()).onSelectDate(date);

    dismiss();
  }

}

File: app/src/main/java/com/rolandvitezhu/todocloud/ui/activity/main/dialogfragment/DatePickerDialogFragment.kt
package com.rolandvitezhu.todocloud.ui.activity.main.dialogfragment

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.Dialog
import android.os.Bundle
import android.widget.DatePicker
import androidx.appcompat.app.AppCompatDialogFragment
import androidx.lifecycle.ViewModelProvider
import com.rolandvitezhu.todocloud.R
import com.rolandvitezhu.todocloud.ui.activity.main.viewmodel.TodosViewModel
import org.threeten.bp.LocalDate

class DatePickerDialogFragment : AppCompatDialogFragment(), OnDateSetListener {

    private val todosViewModel by lazy {
        ViewModelProvider(requireActivity()).get(TodosViewModel::class.java)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val year = todosViewModel.ldDueDate?.year ?: 0
        val month = todosViewModel.ldDueDate?.monthValue ?: 0
        val day = todosViewModel.ldDueDate?.dayOfMonth ?: 0

        return DatePickerDialog(
                requireActivity(),
                R.style.MyPickerDialogTheme,
                this,
                year, month - 1, day
        )
    }

    override fun onDateSet(view: DatePicker, year: Int, month: Int, day: Int) {
        todosViewModel.setTodoDueDate(LocalDate.of(year, month + 1, day))
        dismiss()
    }
}